appserver/web/war-util/src/main/java/org/glassfish/web/loader/WebappClassLoader.java

```diff
@@ -1131,7 +1138,9 @@ public void close() throws IOException {
         if (status == LifeCycleStatus.CLOSED) {
             return;
         }
-        LOG.log(INFO, "close(), this:\n{0}", this);
+        LOG.log(INFO, () -> "close(), " + this.getClass().getSimpleName()
```

Comment:
The original state was very useful for quite regular issues of users who used incompatible dependencies packed directly to war. And it doesn't log too often, just on CL close.

Reply:
It logs one every app deployment. To me, the big INFO message with urls is just noise. INFO messages to me are short statements about what's going on, with a few simple values. Not something that takes the whole screen to show the whole message. That's either CONFIG (which isn't supported by System.Logger), or DEBUG, because it's only relevant in case of issues. In the end, the urls list only shows what's inside `WEB-INF/lib` in the WAR, people can unpack the WAR and see.

Reply:
Knowing simple name of the closed classloader class is worthless. Currently it reports some statistics and also the list of managed libraries, which is very useful in many use cases (missing/duplicit/harmful dependencies, etc). I use it nearly everyday, with some issues detected which were not resolved yet.
We can discuss that with others - if I am the only one who uses it or not.

Btw it logs also the context which is also used for the name of the classloader - and has the same issue with "unknown" value - I would like rather to clear it and make it final rather than increasing number of ways to make it "weird".

Reply:
I admit I was just too lazy to look at other places that use the constructor. It's better it all places pass the app name so that it can be in the log.

But I would still turn this log into the DEBUG level. If you think it's essential for development, then we can make the default log configuration set the log level of the `org.glassfish.web.loader.WebappClassLoader` logger to FINE. But for production, it makes more sense to disable this log with the NORMAL level. Would you agree with that?

Reply:
> just on CL close.

Maybe we need to look a bit at the wording of the log message too.

I mean, it now sounds like a command.

"Close this: [list]"

Which may make the user wonder if they need to close something explicitly, but they have no idea how to do it.

Reply:
Basically the important information should be always logged:

* Name and all endpoints of the application
* Classpaths 
* When components changed their state, are usable or when they were disabled and why

This reports the classpath of the application. Maybe it would be better to report it on start rather than on close, but because the start is bit more complicated, this is now better.